### ISSUE TITLE
SIGSEGV after update to v0.10.14

### ISSUE THREAD

**Issue by pawelrosada** [infracost/infracost#2180]
```
time="2022-12-09T12:12:52Z" level=info msg="Starting: Downloading Terraform modules"
[34](https://github.com/xxx-com/terraform/actions/runs/3657230832/jobs/6180589166#step:6:35)
time="2022-12-09T12:12:52Z" level=info msg="Starting: Downloading Terraform modules"
[35](https://github.com/xxx-com/terraform/actions/runs/3657230832/jobs/6180589166#step:6:36)
time="2022-12-09T12:12:52Z" level=info msg="Starting: Evaluating Terraform directory"
[36](https://github.com/xxx-com/terraform/actions/runs/3657230832/jobs/6180589166#step:6:37)
time="2022-12-09T12:12:52Z" level=info msg="Starting: Retrieving cloud prices to calculate costs"
[37](https://github.com/xxx-com/terraform/actions/runs/3657230832/jobs/6180589166#step:6:38)
time="2022-12-09T12:12:53Z" level=info msg="Starting: Evaluating Terraform directory"
[38](https://github.com/xxx-com/terraform/actions/runs/3657230832/jobs/6180589166#step:6:39)
time="2022-12-09T12:12:53Z" level=info msg="Starting: Retrieving cloud prices to calculate costs"
[39](https://github.com/xxx-com/terraform/actions/runs/3657230832/jobs/6180589166#step:6:40)
time="2022-12-09T12:12:53Z" level=info msg="Starting: Evaluating Terraform directory"
[40](https://github.com/xxx-com/terraform/actions/runs/3657230832/jobs/6180589166#step:6:41)
time="2022-12-09T12:12:54Z" level=info msg="Starting: Retrieving cloud prices to calculate costs"
[41](https://github.com/xxx-com/terraform/actions/runs/3657230832/jobs/6180589166#step:6:42)
time="2022-12-09T12:12:54Z" level=info msg="Starting: Evaluating Terraform directory"
[42](https://github.com/xxx-com/terraform/actions/runs/3657230832/jobs/6180589166#step:6:43)
time="2022-12-09T12:12:55Z" level=info msg="Starting: Retrieving cloud prices to calculate costs"
[43](https://github.com/xxx-com/terraform/actions/runs/3657230832/jobs/6180589166#step:6:44)
time="2022-12-09T12:12:58Z" level=info msg="Starting: Evaluating Terraform directory"
[44](https://github.com/xxx-com/terraform/actions/runs/3657230832/jobs/6180589166#step:6:45)
time="2022-12-09T12:12:58Z" level=info msg="Starting: Evaluating Terraform directory"
[45](https://github.com/xxx-com/terraform/actions/runs/3657230832/jobs/6180589166#step:6:46)
time="2022-12-09T12:13:01Z" level=info msg="Starting: Retrieving cloud prices to calculate costs"
[46](https://github.com/xxx-com/terraform/actions/runs/3657230832/jobs/6180589166#step:6:47)
time="2022-12-09T12:13:02Z" level=warning msg="Skipping resource default_pool. Infracost currently does not support E2 custom instances"
[47](https://github.com/xxx-com/terraform/actions/runs/3657230832/jobs/6180589166#step:6:48)
time="2022-12-09T12:13:02Z" level=warning msg="Skipping resource module.xxx.module.gke.google_container_node_pool.pools[\"db-pool\"]. Infracost currently does not support E2 custom instances"
[48](https://github.com/xxx-com/terraform/actions/runs/3657230832/jobs/6180589166#step:6:49)
time="2022-12-09T12:13:02Z" level=info msg="Starting: Retrieving cloud prices to calculate costs"
[49](https://github.com/xxx-com/terraform/actions/runs/3657230832/jobs/6180589166#step:6:50)
time="2022-12-09T12:13:02Z" level=warning msg="No prices found for module.xxx.module.gke.google_container_node_pool.pools[\"php-cron-spot-pool\"] Instance usage (Linux/UNIX, on-demand, c2d-highcpu-2), using 0.00"
[50](https://github.com/xxx-com/terraform/actions/runs/3657230832/jobs/6180589166#step:6:51)
panic: runtime error: invalid memory address or nil pointer dereference
[51](https://github.com/xxx-com/terraform/actions/runs/3657230832/jobs/6180589166#step:6:52)
[signal SIGSEGV: segmentation violation code=0x1 addr=0x38 pc=0xbe4611]
[52](https://github.com/xxx-com/terraform/actions/runs/3657230832/jobs/6180589166#step:6:53)

[53](https://github.com/xxx-com/terraform/actions/runs/3657230832/jobs/6180589166#step:6:54)
goroutine 645 [running]:
[54](https://github.com/xxx-com/terraform/actions/runs/3657230832/jobs/6180589166#step:6:55)
github.com/infracost/infracost/internal/schema.(*Resource).FlattenedSubResources(0xc001ebd5c0)
[55](https://github.com/xxx-com/terraform/actions/runs/3657230832/jobs/6180589166#step:6:56)
	/home/runner/work/infracost/infracost/internal/schema/resource.go:85 +0xf1
[56](https://github.com/xxx-com/terraform/actions/runs/3657230832/jobs/6180589166#step:6:57)
github.com/infracost/infracost/internal/apiclient.(*PricingAPIClient).batchQueries(0xc0010a7c80?, 0xc001ebd5c0)
[57](https://github.com/xxx-com/terraform/actions/runs/3657230832/jobs/6180589166#step:6:58)
	/home/runner/work/infracost/infracost/internal/apiclient/pricing.go:155 +0x2b1
[58](https://github.com/xxx-com/terraform/actions/runs/3657230832/jobs/6180589166#step:6:59)
github.com/infracost/infracost/internal/apiclient.(*PricingAPIClient).RunQueries(0xc002172320, 0xc001ebd5c0)
[59](https://github.com/xxx-com/terraform/actions/runs/3657230832/jobs/6180589166#step:6:60)
	/home/runner/work/infracost/infracost/internal/apiclient/pricing.go:108 +0x3e
[60](https://github.com/xxx-com/terraform/actions/runs/3657230832/jobs/6180589166#step:6:61)
github.com/infracost/infracost/internal/prices.GetPrices(0xc000e2d680?, 0xc002172320, 0x1?)
[61](https://github.com/xxx-com/terraform/actions/runs/3657230832/jobs/6180589166#step:6:62)
	/home/runner/work/infracost/infracost/internal/prices/prices.go:74 +0x47
[62](https://github.com/xxx-com/terraform/actions/runs/3657230832/jobs/6180589166#step:6:63)
github.com/infracost/infracost/internal/prices.GetPricesConcurrent.func1(0x78e4ca?, 0xc00095c720?)
[63](https://github.com/xxx-com/terraform/actions/runs/3657230832/jobs/6180589166#step:6:64)
	/home/runner/work/infracost/infracost/internal/prices/prices.go:48 +0x5c
[64](https://github.com/xxx-com/terraform/actions/runs/3657230832/jobs/6180589166#step:6:65)
created by github.com/infracost/infracost/internal/prices.GetPricesConcurrent
[65](https://github.com/xxx-com/terraform/actions/runs/3657230832/jobs/6180589166#step:6:66)
	/home/runner/work/infracost/infracost/internal/prices/prices.go:46 +0x9c
[66](https://github.com/xxx-com/terraform/actions/runs/3657230832/jobs/6180589166#step:6:67)
Error: Process completed with exit code 2.
```

I recive SIGSEGV when try to calculate our infrastructure. Before v0.10.14 everything works fine.
Our .infracost.yaml
```
name: Infracost Generate Pricing Diff
on: [pull_request]
env:
  # If you use private modules you'll need this env variable to use
  # the same ssh-agent socket value across all jobs & steps.
  SSH_AUTH_SOCK: /tmp/ssh_agent.sock

jobs:
  multi-project-config-file:
    name: Infracost diff
    runs-on: ubuntu-latest
    permissions:
      contents: read
      pull-requests: write
      id-token: write
    env:
      TF_ROOT: .

    steps:
      # If you use private modules, add an environment variable or secret
      # called GIT_SSH_KEY with your private key, so Infracost can access
      # private repositories (similar to how Terraform/Terragrunt does).
      #      - name: add GIT_SSH_KEY
      #        run: |
      #           ssh-agent -a $SSH_AUTH_SOCK
      #           mkdir -p ~/.ssh
      #           echo "${{ secrets.GIT_SSH_KEY }}" | tr -d '\r' | ssh-add -
      #           ssh-keyscan github.com >> ~/.ssh/known_hosts
      - name: Setup Infracost
        uses: infracost/actions/setup@v2
        # See https://github.com/infracost/actions/tree/master/setup for other inputs
        # If you can't use this action, see Docker images in https://infracost.io/cicd
        with:
          api-key: ${{ secrets.INFRACOST_API_KEY }}

      # Checkout the branch you want Infracost to compare costs against.This example is using the
      # target PR branch.
      - name: Checkout base branch
        uses: actions/checkout@v3
        with:
          ref: '${{ github.event.pull_request.base.ref }}'

      - id: 'auth2'
        uses: 'google-github-actions/auth@v1'
        with:
          workload_identity_provider: ${{ secrets.WORKLOAD_IDENTITY_PROVIDER_ID }}
          service_account: ${{ secrets.SERVICE_ACCOUNT }}
      - name: 'Set up Cloud SDK'
        uses: 'google-github-actions/setup-gcloud@v1'
      # Generate Infracost JSON file as the baseline.
      - name: Generate Infracost cost estimate baseline
        run: |
          infracost breakdown --config-file=${TF_ROOT}/.infracost.yaml \
                              --format=json \
                              --out-file=/tmp/infracost-base.json

      # Checkout the current PR branch so we can create a diff.
      - name: Checkout PR branch
        uses: actions/checkout@v3
      - id: 'auth'
        uses: 'google-github-actions/auth@v1'
        with:
          workload_identity_provider: ${{ secrets.WORKLOAD_IDENTITY_PROVIDER_ID }}
          service_account: ${{ secrets.SERVICE_ACCOUNT }}
      # Generate an Infracost diff and save it to a JSON file.
      - name: Generate Infracost diff
        run: |
          infracost diff --config-file=${TF_ROOT}/.infracost.yaml \
                              --format=json \
                              --compare-to=/tmp/infracost-base.json \
                              --out-file=/tmp/infracost.json

      # Posts a comment to the PR using the 'update' behavior.
      # This creates a single comment and updates it. The "quietest" option.
      # The other valid behaviors are:
      #   delete-and-new - Delete previous comments and create a new one.
      #   hide-and-new - Minimize previous comments and create a new one.
      #   new - Create a new cost estimate comment on every push.
      # See https://www.infracost.io/docs/features/cli_commands/#comment-on-pull-requests for other options.
      - name: Post Infracost comment
        run: |
          infracost comment github --path=/tmp/infracost.json \
                                   --repo=$GITHUB_REPOSITORY \
                                   --github-token=${{github.token}} \
                                   --pull-request=${{github.event.pull_request.number}} \
                                   --behavior=update
```

Crash on stage `  - name: Generate Infracost cost estimate baseline`

Any idea what is going on?
```
Run infracost/actions/setup@v2
10
/usr/bin/tar xz --warning=no-unknown-keyword --overwrite -C /home/runner/work/_temp/bdf5359d-e139-4259-8c98-f1f2360b5f2b -f /home/runner/work/_temp/2568cde8-f979-43ed-a52f-37f42231ca8b
11
Setup Infracost CLI version 0.10.14
12
/home/runner/work/_temp/bdf5359d-e139-4259-8c98-f1f2360b5f2b/infracost configure set api_key ***
13
/home/runner/work/_temp/bdf5359d-e139-4259-8c98-f1f2360b5f2b/infracost configure set currency USD```

**Comment by hugorut** [comment]
Thanks for raising this issue @pawelrosada, from the stack trace this looks like a nil pointer issue with our internal Resource schema. We made some changes in the `0.10.14` release to change some resources to use a new schema format, which can return nil in certain cases. It seems like we've missed a nil check here. I'll raise a fix ASAP and we'll hopefully get a new release out soon. In the meantime, I'd recommend downgrading your actions to `0.10.13` and I'll ping you here when a release is live.

**Comment by pawelrosada** [comment]
@hugorut confirm - after downgrade everything works! :)

**Comment by pawelrosada** [comment]
@hugorut 
```Running multiple projects in parallel, so log-level=info is enabled by default.
Run with INFRACOST_PARALLELISM=1 to disable parallelism to help debugging.

time="2022-12-14T11:31:57+01:00" level=info msg="Evaluating Terraform directory at ./xxx/xxx-playground/dev"
time="2022-12-14T11:31:57+01:00" level=info msg="Evaluating Terraform directory at ./xxx/cloud-tools/sonarqube/non-production"
time="2022-12-14T11:31:57+01:00" level=info msg="Starting: Downloading Terraform modules"
time="2022-12-14T11:31:57+01:00" level=info msg="Starting: Evaluating Terraform directory"
time="2022-12-14T11:31:57+01:00" level=info msg="Starting: Downloading Terraform modules"
time="2022-12-14T11:31:57+01:00" level=info msg="Evaluating Terraform directory at ./xxx/prod"
time="2022-12-14T11:31:57+01:00" level=info msg="Evaluating Terraform directory at ./xxx/cicd"
time="2022-12-14T11:31:57+01:00" level=info msg="Starting: Evaluating Terraform directory"
time="2022-12-14T11:31:57+01:00" level=info msg="Evaluating Terraform directory at ./xxx/dev"
time="2022-12-14T11:31:57+01:00" level=info msg="Evaluating Terraform directory at ./xxx-terraform/prod"
time="2022-12-14T11:31:57+01:00" level=info msg="Starting: Downloading Terraform modules"
time="2022-12-14T11:31:57+01:00" level=info msg="Starting: Evaluating Terraform directory"
time="2022-12-14T11:31:57+01:00" level=info msg="Starting: Downloading Terraform modules"
time="2022-12-14T11:31:57+01:00" level=info msg="Starting: Evaluating Terraform directory"
time="2022-12-14T11:31:57+01:00" level=info msg="Starting: Downloading Terraform modules"
time="2022-12-14T11:31:57+01:00" level=info msg="Starting: Evaluating Terraform directory"
time="2022-12-14T11:31:57+01:00" level=info msg="Starting: Downloading Terraform modules"
time="2022-12-14T11:31:57+01:00" level=info msg="Starting: Evaluating Terraform directory"
time="2022-12-14T11:31:57+01:00" level=info msg="Starting: Retrieving cloud prices to calculate costs"
time="2022-12-14T11:31:57+01:00" level=info msg="Starting: Retrieving cloud prices to calculate costs"
time="2022-12-14T11:31:57+01:00" level=info msg="Starting: Retrieving cloud prices to calculate costs"
time="2022-12-14T11:31:58+01:00" level=info msg="Starting: Retrieving cloud prices to calculate costs"
time="2022-12-14T11:31:58+01:00" level=info msg="Starting: Retrieving cloud prices to calculate costs"
time="2022-12-14T11:31:59+01:00" level=warning msg="Skipping resource default_pool. Infracost currently does not support E2 custom instances"
time="2022-12-14T11:31:59+01:00" level=warning msg="Skipping resource module.xxx.module.gke.google_container_node_pool.pools[\"db-pool\"]. Infracost currently does not support E2 custom instances"
time="2022-12-14T11:31:59+01:00" level=info msg="Starting: Retrieving cloud prices to calculate costs"
time="2022-12-14T11:31:59+01:00" level=warning msg="No prices found for module.xxx.module.gke.google_container_node_pool.pools[\"php-cron-spot-pool\"] Instance usage (Linux/UNIX, on-demand, c2d-highcpu-2), using 0.00"
time="2022-12-14T11:31:59+01:00" level=warning msg="No prices found for module.xxx.module.gke.google_container_node_pool.pools[\"php-frontend-on-demand-pool\"] Instance usage (Linux/UNIX, on-demand, c2d-highcpu-2), using 0.00"
time="2022-12-14T11:31:59+01:00" level=warning msg="No prices found for module.xxx.module.gke.google_container_node_pool.pools[\"php-frontend-spot-pool\"] Instance usage (Linux/UNIX, on-demand, c2d-highcpu-2), using 0.00"
time="2022-12-14T11:32:00+01:00" level=warning msg="No prices found for module.xxx.module.gke.google_container_node_pool.pools[\"php-cron-on-demand-pool\"] Instance usage (Linux/UNIX, on-demand, c2d-highcpu-2), using 0.00"
time="2022-12-14T11:32:00+01:00" level=info msg="Output saved to /tmp/infracost-base.json"
```
It's works fine! I test run on 0.14 and next your fix. On 0.14 crash, on your fix output like i pasted ;)

**Comment by hugorut** [comment]
Hey @pawelrosada this fix is out in `0.10.15` release, so if you upgrade your actions this should resolve your runtime panic. Thanks for reporting this issue.
